src/service/video-manager-impl.js

```diff
@@ -959,6 +932,369 @@ class VideoEntry {
 }
 
 
+
+/**
+ * @struct @typedef {{
+ *  videoEntry: !VideoEntry,
+ *  video: !../video-interface.VideoInterface,
+ * }}
+ */
+let R82FullscreenEntryDef;
+
+
+/** @private @const {string} */
+const R82_FULLSCREEN_ID_PROP = '__AMP_R82F_ID__';
+
+
+/** @private @const {string} */
+const CENTERED_PORTRAIT = 'i-amphtml-centered-portrait';
+
+
+/** @private @const {string} */
+const CENTERED_LANDSCAPE = 'i-amphtml-centered-landscape';
+
+
+/** Manages rotate-to-fullscreen video. */
+class R82FullscreenManager {
+
+  /** @param {!./ampdoc-impl.AmpDoc} ampdoc */
+  constructor(ampdoc) {
+    const {win} = ampdoc;
+
+    /** @private @const {!./ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
+
+    /** @private {number} */
+    this.nextId_ = 0;
+
+    /** @private @const {!Window} */
+    this.win_ = win;
+
+    /** @private @const {!./timer-impl.Timer} */
+    this.timer_ = Services.timerFor(win);
+
+    /** @private {?R82FullscreenEntryDef} */
+    this.currently_ = null;
+
+    /**
+     * Maps rotate-to-fullscreen entry id to entry.
+     * @private @const {!Object<string, !R82FullscreenEntryDef>}
+     */
+    this.entries_ = {};
+
+    /** @private @const {function()} */
+    const installOrientationObserver =
+        this.installOrientationObserver_.bind(this);
+    this.installOrientationObserver_ = once(installOrientationObserver);
```

Comment:
do we need `once` here? given `R82FullscreenManager` also shouldn't be instantiated more than once.